File: Source/Bifrost.JavaScript.Specs/utils/for_StringMapping/when_resolving_a_string.js
describe("when resolving a string", function () {
    var expectedResult = "Say/hello/to/mr.potatohead";
    var mapping = Bifrost.StringMapping.create({
        format: "{something}/{else}",
        mappedFormat: "Say/{else}/to/{something}"
    });
    var result = mapping.resolve("mr.potatohead/hello");

    it("should expand input string to the mapped string", function () {
        expect(result).toBe(expectedResult);
    });
});

File: Source/Bifrost.Specs/Utils/for_StringMapping/when_resolving_a_string.cs
using Bifrost.Utils;
using Machine.Specifications;
namespace Bifrost.Specs.Utils.for_StringMapping
{
    public class when_resolving_a_string
    {
        const string expected_result = "Say/hello/to/mr.potatohead";
        static StringMapping    mapping;
        static string result;

        Establish context = () => mapping = new StringMapping(
                "{something}/{else}",
                "Say/{else}/to/{something}"
            );

        Because of = () => result = mapping.Resolve("mr.potatohead/hello");

        It should_expand_input_string_to_mapped_string = () => result.ShouldEqual(expected_result);
    }
}
